packages/react-components/react-dialog/src/components/Dialog/Dialog.types.ts

```diff
@@ -1,17 +1,80 @@
+import type * as React from 'react';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import type { DialogContextValue } from '../../contexts/dialogContext';
 
 export type DialogSlots = {
-  root: Slot<'div'>;
+  /**
+   * Dimmed background of dialog.
+   * The default overlay is rendered as a `<div>` with styling.
+   * This slot expects a `<div>` element which will replace the default overlay.
+   * The overlay should have `aria-hidden="true"`.
+   */
+  overlay?: Slot<'div'>;
 };
 
-/**
- * Dialog Props
- */
-export type DialogProps = ComponentProps<DialogSlots>;
+export type DialogOpenChangeEvent = React.KeyboardEvent | React.MouseEvent | KeyboardEvent;
 
-/**
- * State used in rendering Dialog
- */
-// TODO: Add union of props to pick from DialogProps once they're implemented.
-// i.e. Required<Pick<DialogProps, 'property1' | 'property2'>>;
-export type DialogState = ComponentState<DialogSlots>;
+export type DialogOpenChangeData =
+  | { type: 'escapeKeyDown'; open: boolean; event: React.KeyboardEvent }
+  /**
+   * document escape keydown defers from internal escape keydown events because of the synthetic event API
+   */
+  | { type: 'documentEscapeKeyDown'; open: boolean; event: KeyboardEvent }
+  | { type: 'overlayClick'; open: boolean; event: React.MouseEvent }
+  | { type: 'triggerClick'; open: boolean; event: React.MouseEvent };
+
+export type DialogModalType = 'modal' | 'non-modal' | 'alert';
+
+export type DialogContextValues = {
+  dialog: DialogContextValue;
+};
+
+export type DialogProps = ComponentProps<Partial<DialogSlots>> & {
+  /**
+   * Dialog variations.
+   *
+   * `modal`: When this type of dialog is open, the rest of the page is dimmed out and cannot be interacted with.
+   * The tab sequence is kept within the dialog and moving the focus outside
+   * the dialog will imply closing it. This is the default type of the component.
+   *
+   * `non-modal`: When a non-modal dialog is open, the rest of the page is not dimmed out
+   * and users can interact with the rest of the page.
+   * This also implies that the tab focus can move outside the dialog when it reaches the last focusable element.
+   *
+   * `alert`: is a special type of modal dialogs that interrupts the user's workflow
+   * to communicate an important message or ask for a decision.
+   * Unlike a typical modal dialog, the user must take an action through the options given to dismiss the dialog,
+   * and it cannot be dismissed through the dimmed background or escape key.
+   *
+   * @default modal
+   */
+  modalType?: DialogModalType;
+  /**
+   * Controls the open state of the dialog
+   * @default false
+   */
+  open?: boolean;
+  /**
+   * Default value for the uncontrolled open state of the dialog.
+   * @default false
+   */
+  defaultOpen?: boolean;
```

Comment:
just an FYI about the `defaultOpen` stuff in SSR: https://github.com/microsoft/fluentui/pull/23612